arangod/Cluster/ActionBase.h

```diff
@@ -112,6 +112,14 @@ class ActionBase {
   /// @brief finalize statistics
   void endStats();
 
+  /**
+   *  @brief    update progress by long running processes, while allowing to
+   *            abort the process by user.
+   *  @param  u percentage of work done
+   *  @return   abort if !ok(), true if ok(), with reason to abort.
+   */
+  virtual arangodb::Result setProgress(uint64_t u);
```

Comment:
I don't understand why Result needed now.
Could you explain for me please?

Reply:
Again, this is not the scope of this PR. There is however a possibility, why such a prospective function impl might fail.